exp/internal/controllers/machinepool_controller_phases.go

```diff
@@ -537,3 +570,29 @@ func (r *MachinePoolReconciler) waitForMachineCreation(ctx context.Context, mach
 
 	return nil
 }
+
+func (r *MachinePoolReconciler) getNodeRefMap(ctx context.Context, c client.Client) *getNodeRefMapResult {
+	log := ctrl.LoggerFrom(ctx)
+	nodeRefsMap := make(map[string]corev1.Node)
```

Comment:
```suggestion
	nodeRefsMap := make(map[string]*corev1.Node)
```
I know it was the same before, but I would prefer making this a pointer. This makes it a lot easier to distinguish between a Node and nil (currently we pass an empty Node struct around in some cases)